src/layers/Layer.js

```diff
@@ -57,8 +51,9 @@ export default class Layer extends Observable {
 
     if (this.olLayer) {
       this.olLayer.setVisible(this.visible);
-      if (this.zIndex) {
-        this.olLayer.setZIndex(this.zIndex);
+      if (zIndex) {
+        this.properties.zIndex = zIndex;
```

Comment:
Why do we need this twice?

Reply:
Mario wants this to be written in the properties so it is accessible via Layer.get('zIndex')

Reply:
This smells like trouble. What happens if you write `Layer.set('zIndex', 999)`? You end up with two different values.

Why not access this using `Layer.olLayer.getZIndex()`?

Reply:
Good point, but your example already applied before my changes

Reply:
Will mention this to Mario